compaction/time_window_compaction_strategy.cc

```diff
@@ -39,7 +39,13 @@ time_window_compaction_strategy_options::time_window_compaction_strategy_options
         window_unit = valid_window_units_it->second;
     }
 
-    it = options.find(COMPACTION_WINDOW_SIZE_KEY);
+    return window_unit;
+}
+
+int time_window_compaction_strategy_options::validate_compaction_window_size(const std::map<sstring, sstring>& options) {
```

Comment:
I wonder if it's possible to pass a mutable `std::map<sstring, sstring>& options`
and to erase the relevant option once it's validated
so we can check at the end if there are unrecognized options that are left.
It's possible that due to a typo, the user means to set some options but the code doesn't recognize it and it might go unnoticed.
(I guess the "nested" size_tiered options will need to handled too)

Reply:
The "nested" options are checked in the following commits. The whole validation has the structure:
- `validate_*` helpers validate a single option
- `*_compaction_strategy_options::validate` validate options specific for the respective strategy (and use ^)
- `compaction_strategy_impl::validate_options` validates options that are common for each strategy type
- `*_compaction_strategy::validate_options` validate "nested" options for each strategy (and use ^ and ^^)